src/components/fidelity_bond/AccountSelector.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState, useMemo, useCallback } from 'react'
+import * as rb from 'react-bootstrap'
+
+import { Account } from '../../context/WalletContext'
+// @ts-ignore
+import { useSettings } from '../../context/SettingsContext'
+import AccountCheckbox, { SelectableAccount } from './AccountCheckbox'
+
+interface AccountSelectorProps {
+  accounts: SelectableAccount[]
+  type?: 'checkbox' | 'radio'
+  onChange: (selectedAccounts: Account[]) => void
+  displayDisabledAccounts?: boolean
+}
+const AccountSelector = ({
+  accounts,
+  type = 'radio',
+  onChange,
+  displayDisabledAccounts = true,
+}: AccountSelectorProps) => {
+  const settings = useSettings()
+  const [selected, setSelected] = useState<Account[]>([])
+
+  const selectableAccounts = useMemo(() => {
+    return accounts.filter((it) => !it.disabled)
+  }, [accounts])
+
+  const totalAmount = useMemo(() => {
+    return accounts
+      .map((it) => it.utxos.reduce((acc, curr) => acc + curr.value, 0))
+      .reduce((acc, curr) => acc + curr, 0)
+  }, [accounts])
```

Comment:
Is there a benefit of recalculating this vs. taking it from `currentWallet`?

Reply:
I don't know exactly what you mean. The actual total value in sats is not stored anywhere. I was about to add a total value in sats to the `useBalance` hook (think there is a TODO in the comments iirc).
But in this case, the accounts can be provided and it felt wrong to bring in a total balance when it can easily be calculated from the input values. Furthermore, total value in btc is in `WalletDisplayInfo`.. however, the current `satsToBtc` util function is not to be trusted.  It parses a float string and does rounding.. nothing good can ever come from that. It is all good and well when the outputs are used for displaying stuff (notice the `Display` in the name).. For Fidelity Bonds, I did not want to take the chance.

Or maybe I misunderstood and you had something completely different in mind. Can you elaborate?

Reply:
Yeah I meant `WalletDisplayInfo`. I mean it doesn't really matter here probably but it's a red flag that goes off in my head when there's two different sources of some value since then there's the (at least theoretical) possibility of  stuff getting out of sync. That's why I raised it but you can best judge if it's ok to do it like this or somehow get access to `WalletDisplayInfo`. Up to you! 😄